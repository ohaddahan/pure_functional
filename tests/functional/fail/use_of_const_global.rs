use functional_macro::pure_functional;

const GLOBAL: u32 = 1;
#[pure_functional]
fn test(i1: u32) -> u32 {
    i1 + GLOBAL
}

fn main() {
    let i1 = 5;
    let result = test(i1);
    assert_eq!(result, i1 + GLOBAL);
}
