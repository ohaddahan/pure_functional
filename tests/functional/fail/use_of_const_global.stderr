error[E0425]: cannot find value `GLOBAL` in this scope
 --> tests/functional/fail/use_of_const_global.rs:6:10
  |
6 |     i1 + GLOBAL
  |          ^^^^^^ not found in this scope
  |
  = help: consider importing this constant:
          crate::GLOBAL
